framework/test/unit/modules/random/endpoint.spec.ts

```diff
@@ -359,23 +336,67 @@ describe('RandomModuleEndpoint', () => {
 			expect(hasHashOnion.remaining).toBe(0);
 		});
 
-		it('should return hasHashOnion true with valid remaining', async () => {
+		it('should return hasSeed true with valid number of remaining hashes', async () => {
+			// Arrange
+			const { address } = genesisValidators.validators[0];
+			const usedCount = 20;
+
+			await randomEndpoint.setHashOnion({ ...context, params: { address, count, distance } });
+
+			const usedHashOnionStore = randomEndpoint['offchainStores'].get(UsedHashOnionsStore);
+			await usedHashOnionStore.set(
+				context,
+				cryptography.address.getAddressFromLisk32Address(address),
+				{
+					usedHashOnions: [{ count: usedCount, height: 2121 }],
+				},
+			);
+
+			// Act
 			const hasHashOnion = await randomEndpoint.hasHashOnion({ ...context, params: { address } });
 
 			// Assert
 			expect(hasHashOnion.hasSeed).toBe(true);
-			expect(hasHashOnion.remaining).toEqual(1000 - 20);
+			expect(hasHashOnion.remaining).toBe(count - usedCount);
 		});
 
-		it('should return hasHashOnion true with remaining the same as original when usedHashOnions does not exist', async () => {
+		it('should return hasSeed true with all remaining hashes when usedHashOnions does not exist', async () => {
+			// Arrange
+			const { address } = genesisValidators.validators[1];
+			await randomEndpoint.setHashOnion({ ...context, params: { address, count, distance } });
+
+			// Act
 			const hasHashOnion = await randomEndpoint.hasHashOnion({
 				...context,
-				params: { address: address2 },
+				params: { address },
 			});
 
 			// Assert
 			expect(hasHashOnion.hasSeed).toBe(true);
-			expect(hasHashOnion.remaining).toBe(1000);
+			expect(hasHashOnion.remaining).toBe(count);
+		});
+
+		it('should return hasSeed false with 0 remaining hashes when a hash onion is used up', async () => {
```

Comment:
```suggestion
		it('should return hasSeed false with 0 remaining hashes when all hash onions are used up', async () => {
```
If I understand correctly this case, all hash onions are used up, not just one ?